features/theme/index.tsx

```diff
@@ -121,7 +123,12 @@ export function createTheme({
         );
 
         React.useEffect(() => {
-            Colors.setScheme(scheme as SchemeType);
+            if (ignoreSystemMode) {
+                Colors.setScheme(scheme as SchemeType);
+            } else {
+                const systemScheme = Appearance.getColorScheme() as 'light' | 'dark' | null;
+                Colors.setScheme(systemScheme || 'default');
+            }
```

Comment:
Since lines 126-131 are repeated of lines 89-95, can you write a common function to handle both, please?